### DISCUSSION TITLE
The workflow is not valid. .github/workflows/main.yml: Expected format {org}/{repo}[/path]@ref. Actual 'actions/checkout' Input string was not in a correct format

### DISCUSSION THREAD

**Discussion by GavinCameron** [community/community#25655] [Actions]
<p>I built a container action in a repo:</p>
<p><code>ourorg/generic-snowsql-action</code></p>
<p>This repo contains an <code>action.yml</code> file. This repo is intended to be a generic container action that I can recycle in github actions workflows in other repos. E.g. I created another repo <code>ourorg/specific-task-that-will-use-snowsql</code> and a workflow and I would like to reference the container action from the other repo <code>ourorg/generic-snowsql-action</code>.</p>
<p>When I attempt to checkout this container action I get an error message:</p>
<blockquote>
<p><code>Check failure on line 9 in .github/workflows/main.yml. The workflow is not valid. .github/workflows/main.yml (Line: 9, Col: 15): Expected format {org}/{repo}[/path]@ref. Actual 'actions/checkout' Input string was not in a correct format.</code></p>
</blockquote>
<p>Here is <code>ourorg/specific-task-that-will-use-snowsql/.github/workflows/main.yml</code></p>
<pre><code class="lang-auto">on: [push]

jobs:
  ssql_job:
    runs-on: ubuntu-latest
    name: Build ssql image
    steps:
      - name: Checkout # To use this repository's private action, you must check out the repository
        uses: actions/checkout
        with:
          repository: ourorg/generic-snowsql-action
          ref: main
      - name: Build
        uses: ./ # Uses an action in the root directory 
        with:
          config: ./config # the config file for snowsql
          query: ./training-data.sql
</code></pre>
<p>When I push this I get the error above. Presumably there’s an error with this piece, line 9 is</p>
<pre><code class="lang-auto">    steps:
      - name: Checkout # To use this repository's private action, you must check out the repository
        uses: actions/checkout # this is line 9!
</code></pre>
<p>How can I correctly tell my main.yml to checkout the container action I created over at <code>ourorg/generic-snowsql-action</code>?</p>

**Comment by airtower-luna** [comment]
<p>This is the key part of the error message:</p>
<blockquote>
<p><code>Expected format {org}/{repo}[/path]@ref</code></p>
</blockquote>
<p><code>actions/checkout</code> misses the <code>@ref</code> part, which is mandatory. It must provide the ref in the <a href="https://github.com/actions/checkout"><code>actions/checkout</code> repository</a> to use, basically the version of the action. <code>ref</code> can be a commit SHA, tag, or branch, depending on your security and update needs.</p>

**Comment by GavinCameron** [comment]
<p>Hi <a class="mention" href="/airtower-luna">@airtower-luna</a> thanks for the info, I like the idea of using a commit sha and I updated my line from this:</p>
<p><code>repository: ourorg/ds-ssql-gh-action</code></p>
<p>To this:</p>
<p><code>repository: ourorg/ds-ssql-gh-action@10c0fabc8ec358d5a2c7b0ab3fdce38c195ff55b</code></p>
<p>I got the exact same error message :confused:</p>
<p>While I’m learning this I didn’t want to use a version tag. Is the syntax incorrect here for using a commit sha?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The error message isn’t about the <code>repository</code> input to <code>actions/checkout</code>, it’s about the <code>uses: actions/checkout</code> line. You need to add the ref <em>of actions/checkout</em> that you want to use.</p>
